web/src/stream_pill.ts

```diff
@@ -34,7 +34,9 @@ export function create_item_from_stream_name(
         return undefined;
     }
 
-    const existing_ids = current_items.map((item) => item.stream_id);
+    const existing_ids = current_items
+        .flatMap((item) => (item.type === "stream" ? item : []))
+        .map((item) => item.stream_id);
     if (existing_ids.includes(sub.stream_id)) {
```

Comment:
These extra array operations, and others like them, are completely unnecessary for this use.

- #30080